skyportal/utils/gcn.py

```diff
@@ -0,0 +1,254 @@
+    if retraction is not None:
+        retraction = int(retraction.attrib['value'])
+        if retraction == 1:
+            return None
+
+    # Try error cone
+    loc = root.find('./WhereWhen/ObsDataLocation/ObservationLocation')
+    if loc is None:
+        return None
+
+    ra = loc.find('./AstroCoords/Position2D/Value2/C1')
+    dec = loc.find('./AstroCoords/Position2D/Value2/C2')
+    error = loc.find('./AstroCoords/Position2D/Error2Radius')
+
+    if None in (ra, dec, error):
+        return None
+
+    ra, dec, error = float(ra.text), float(dec.text), float(error.text)
+
+    # Apparently, all experiments *except* AMON report a 1-sigma error radius.
+    # AMON reports a 90% radius, so for AMON, we have to convert.
+    if mission == 'AMON':
+        error /= scipy.stats.chi(df=2).ppf(0.95)
+
+    return from_cone(ra, dec, error)
+
+
+def from_cone(ra, dec, error):
+    localization_name = "%.5f_%.5f_%.5f" % (ra, dec, error)
+
+    center = SkyCoord(ra * u.deg, dec * u.deg)
+    radius = error * u.deg
+
+    # Determine resolution such that there are at least
+    # 16 pixels across the error radius.
+    hpx = HEALPix(
+        pixel_resolution_to_nside(radius / 16, round='up'), 'nested', frame=ICRS()
+    )
+
+    # Find all pixels in the 4-sigma error circle.
+    ipix = hpx.cone_search_skycoord(center, 4 * radius)
+
+    # Convert to multi-resolution pixel indices and sort.
+    uniq = ligo.skymap.moc.nest2uniq(nside_to_level(hpx.nside), ipix.astype(np.int64))
+    i = np.argsort(uniq)
+    ipix = ipix[i]
+    uniq = uniq[i]
+
+    # Evaluate Gaussian.
+    distance = hpx.healpix_to_skycoord(ipix).separation(center)
+    probdensity = np.exp(
+        -0.5 * np.square(distance / radius).to_value(u.dimensionless_unscaled)
+    )
+    probdensity /= probdensity.sum() * hpx.pixel_area.to_value(u.steradian)
+
+    skymap = {
+        'localization_name': localization_name,
+        'uniq': uniq.tolist(),
+        'probdensity': probdensity.tolist(),
+    }
+
+    return skymap
+
+
+def from_url(url):
+    def get_col(m, name):
+        try:
+            col = m[name]
+        except KeyError:
+            return None
+        else:
+            return col.tolist()
+
+    filename = os.path.basename(urlparse(url).path)
+
+    skymap = ligo.skymap.io.read_sky_map(url, moc=True)
+
+    skymap = {
+        'localization_name': filename,
+        'uniq': get_col(skymap, 'UNIQ'),
+        'probdensity': get_col(skymap, 'PROBDENSITY'),
+        'distmu': get_col(skymap, 'DISTMU'),
+        'distsigma': get_col(skymap, 'DISTSIGMA'),
+        'distnorm': get_col(skymap, 'DISTNORM'),
+    }
+
+    return skymap
+
+
+def get_contour(localization):
+
+    # Calculate credible levels.
+    prob = localization.flat_2d
+    cls = 100 * ligo.skymap.postprocess.find_greedy_credible_levels(prob)
+
+    # Construct contours and return as a GeoJSON feature collection.
+    levels = [50, 90]
```

Comment:
Same question regarding configurability

Reply:
Same as above. Never say never, but I think its ok if that ends up being a small PR down the line.